src/main/js/controls/Gantt/helpers/translateHelpers.js

```diff
@@ -121,14 +124,32 @@ const getTrackLabelTransformProperty = (scale, config) => (trackName) =>
  * @private
  * @param {object} config - config object derived from input JSON
  * @param {d3.selection} canvasSVG - d3 selection node of canvas svg
- * @returns {object} d3 svg path
+ * @returns {undefined} - returns nothing
  */
-const translateDefs = (config, canvasSVG) =>
+const translateDefs = (config, canvasSVG) => {
     canvasSVG
         .select(`clipPath#${config.clipPathId}`)
         .selectAll("rect")
         .attr("height", getYAxisHeight(config))
         .attr("width", getXAxisWidth(config));
+    if (
+        config.settingsDictionary.shouldCreateDatelineDefs &&
+        config.dateline.length > 0
+    ) {
+        const datelineIndicatorHeight = Math.floor(
+            getDatelineIndicatorHeight() / 2
```

Comment:
Although this is good for maintaining code. This wont help us with the performance. Let me describe why.
* Add a console.log and try resizing the page, notice how many times this function is called? 
* Calling getBoundingClientRect and getBBox essentially causes a force page reflow. 
* Calling it in a resize event is not advised since it would cause layout thrashing
More information here: https://developers.google.com/web/fundamentals/performance/rendering/avoid-large-complex-layouts-and-layout-thrashing

But it is also true that during panning there maybe another dateline with an indicator that has a different height. So I dont see a way out of this. 
Can you add this comment in the JSDocs:
```
* @description
 * Calling getDatelineIndicatorHeight() will trigger a page reflow and resizing the page might cause Layout Thrashing.
 * We understand this and deem it necessary to calculate the indicator height when a new dataset/set of contents are loaded during Panning.
 * Furthermore, this function is called only when panning is enabled and there is a dateline defs element is present.
```

Reply:
ok sure I will add comment